uaclient/entitlements/esm.py

```diff
@@ -36,6 +39,42 @@ def disable(
             update_apt_and_motd_messages(self.cfg)
         return disable_performed, fail
 
+    def setup_local_esm_repo(self) -> None:
+        series = system.get_platform_info()["series"]
+        # Ugly? Yes, but so is python < 3.8 without removeprefix
+        esm_name = self.name[4:]
+        repo_filename = os.path.normpath(
+            ESM_APT_ROOTDIR + self.repo_list_file_tmpl.format(name=self.name),
```

Comment:
Because if any of the paths start with `/`, then `os.path.join` will consider that as an absolute path and ignore the others. Given here we want to join `/var/lib/ubuntu-advantage/apt-esm/` and `/etc` or `/var`, it was easier implemented like this. I don't think this one is worth changing.

Reply:
Ah, I've just learned something about `os.path.join()` today. Thanks!